modules/linear_boltzmann_solvers/a_lbs_solver/source_functions/source_function.cc

```diff
@@ -212,5 +210,45 @@ SourceFunction::AddPointSources(const LBSGroupset& groupset,
       }   // for subscriber
 }
 
+void
+SourceFunction::AddDistributedSources(const LBSGroupset& groupset,
+                                      std::vector<double>& q,
+                                      const std::vector<double>&,
+                                      const SourceFlags source_flags)
+{
+  const bool apply_fixed_src = source_flags & APPLY_FIXED_SOURCES;
+
+  const auto& grid = lbs_solver_.Grid();
+  const auto& discretization = lbs_solver_.SpatialDiscretization();
+  const auto& cell_transport_views = lbs_solver_.GetCellTransportViews();
+  const auto num_groups = lbs_solver_.NumGroups();
+
+  const auto gs_i = groupset.groups_.front().id_;
+  const auto gs_f = groupset.groups_.back().id_;
+
+  // Go through each distributed source, and its subscribing cells
+  if (not lbs_solver_.Options().use_src_moments and apply_fixed_src)
```

Comment:
I don't know that we have a style guide for this, but I think adding braces here and for the immediately following loop makes the code much easier to understand.